src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -110,11 +137,50 @@ public void storeMessage(Message message) {
     return messages;
   }
 
+  /*
+   * Constructs a list of markers with all the marker entities
+   */
+  private List<Marker> getMarkersHelperFunction(Query query){
+    List<Marker> markers = new ArrayList<>();
+
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      try {
+        String key = entity.getKey().getName();
+        double lat = (double) entity.getProperty("lat");
+        double lng = (double) entity.getProperty("lng");
+        String content = (String) entity.getProperty("content");
+
+        Marker marker = new Marker(lat, lng, content, key);
+        markers.add(marker);
+      } catch (Exception e) {
+        System.err.println("Error reading message.");
+        System.err.println(entity.toString());
+        e.printStackTrace();
+      }
+    }
+    return markers;
+  }
+
+  /* Returns all the message entities based on the parentKey */
+  public List<Message> getMessagesForParentKey(String key) {
+    List<Message> messages;
+
+    Query query =
+        new Query("Message")
+            .setFilter(new Query.FilterPredicate("parentKey", FilterOperator.EQUAL, key))
```

Comment:
Querying by this way works, but its not exactly following how Datastore is meant to be used. This way of querying does not take advantage of datastore's ancestor filters in queries. This just does a normal field filter.

I guess this works, but if that's the case, we shouldn't name it parentKey (because that makes it sound like we are using the actual Datastore parent key). We should name it something like `tag`.